src/redis/cache_impl.go

```diff
@@ -210,6 +214,27 @@ func (this *rateLimitCacheImpl) DoLimit(
 	return responseDescriptorStatuses
 }
 
+func CalculateReset(currentLimit *pb.RateLimitResponse_RateLimit) *duration.Duration {
+	sec := unitInSeconds(currentLimit.Unit)
+	now := limiter.NewTimeSourceImpl().UnixNow()
+	return &duration.Duration{Seconds: sec - now%sec}
+}
+
+func unitInSeconds(unit pb.RateLimitResponse_RateLimit_Unit) int64 {
```

Comment:
We already have this function here: https://github.com/envoyproxy/ratelimit/blob/e91321b10f1ad7691d0348e880bd75d0fca05758/src/limiter/cache_key.go#L39

Can you factor it into a utility?